crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1271,68 +1193,103 @@ where
+                        Ok(None) => ReduceOutput::Continue(state),
+                        Err(err) => {
+                            let concise_name = name.concise();
+                            debug!(?tx_digest, name=?concise_name, "Error processing certificate from validator: {:?}", err);
+                            self.metrics
+                                .process_cert_errors
+                                .with_label_values(&[&concise_name.to_string(), err.as_ref()])
+                                .inc();
+                            state.errors.push((err, vec![name], weight));
+                            state.bad_stake += weight;
+                            if state.bad_stake > validity {
+                                ReduceOutput::Failed(state)
+                            } else {
+                                ReduceOutput::Continue(state)
                             }
                         }
-                        ReduceOutput::Continue(state)
-                    })
-                },
-                // A long timeout before we hear back from a quorum
-                self.timeouts.pre_quorum_timeout,
-            )
-            .await.map_err(|state| {
+                    }
+                })
+            },
+            // A long timeout before we hear back from a quorum
+            self.timeouts.pre_quorum_timeout,
+        )
+        .await
+        .map_err(|state| {
+            debug!(
+                ?tx_digest,
+                num_unique_effects = state.effects_map.unique_key_count(),
+                bad_stake = state.bad_stake,
+                "Received effects responses from validators"
+            );
+            QuorumExecuteCertificateError {
+                total_stake: self.committee.total_votes,
+                errors: state.errors,
+            }
+        })
+    }
+
+    fn handle_process_certificate_response(
+        &self,
+        tx_digest: &TransactionDigest,
+        state: &mut ProcessCertificateState,
+        response: SuiResult<VerifiedHandleCertificateResponse>,
+        name: AuthorityName,
+    ) -> SuiResult<Option<VerifiedCertifiedTransactionEffects>> {
+        match response {
+            Ok(VerifiedHandleCertificateResponse { signed_effects }) => {
                 debug!(
                     ?tx_digest,
-                    num_unique_effects = state.effects_map.unique_key_count(),
-                    bad_stake = state.bad_stake,
-                    "Received effects responses from validators"
+                    name = ?name.concise(),
+                    "Validator handled certificate successfully",
                 );
-                QuorumExecuteCertificateError {
-                    total_stake: self.committee.total_votes,
-                    errors: state.errors,
+                let signed_effects = signed_effects.into_inner();
+                // Note: here we aggregate votes by the hash of the effects structure
```

Comment:
We are not hashing if i read the code correctly

Reply:
We are using the the digest of the effects, which is the hash?